tests/capture/test_capture_cond.py

```diff
@@ -516,3 +611,88 @@ def test_circuit_consts(self, pred, arg, expected):
         jaxpr = jax.make_jaxpr(circuit_with_consts)(*args)
         res_ev_jxpr = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
         assert np.allclose(res_ev_jxpr, expected), f"Expected {expected}, but got {res_ev_jxpr}"
+
+    @pytest.mark.parametrize("reset", [True, False])
+    @pytest.mark.parametrize("postselect", [None, 0, 1])
+    @pytest.mark.parametrize("shots", [None, 20])
+    def test_mcm_predicate_execution(self, reset, postselect, shots):
+        """Test that QNodes executed with mid-circuit measurement predicates for
+        qml.cond give correct results."""
+        device = qml.device("default.qubit", wires=3, shots=shots, seed=jax.random.PRNGKey(1234))
+
+        def true_fn(arg):
+            qml.RX(arg, 0)
+
+        def false_fn(arg):
+            qml.RY(arg, 0)
+
+        @qml.qnode(device)
+        def f(x, y):
+            qml.RX(x, 0)
+            m = qml.measure(0, reset=reset, postselect=postselect)
+
+            qml.cond(m, true_fn, false_fn)(y)
+            return qml.expval(qml.Z(0))
+
+        params = [2.5, 4.0]
+        res = f(*params)
+        qml.capture.disable()
+        expected = f(*params)
+
+        assert np.allclose(res, expected), f"Expected {expected}, but got {res}"
```

Comment:
Wait, how is this test succeeding with 20 shots and default precision? :thinking: :sweat_smile:

Reply:
Seeding 🙌

Reply:
o.O That's crazy. But oh well...